util/util.go

```diff
@@ -150,9 +151,9 @@ func IsValidHash(s string) bool {
 
 // XOR takes two byte slices, XORs them together, returns the resulting slice.
 func XOR(a, b []byte) []byte {
+	_ = b[len(a)-1] // keeping same behaviour as previously but this looks like a bug
```

Comment:
We should revisit this later.

Reply:
AFAIK this is only ever called with slices of the same length, so we could check that `if len(a) != len(b) { panic() }`.